ydb/library/yql/minikql/mkql_node.cpp

```diff
@@ -1726,15 +1726,15 @@ bool TCallableType::IsConvertableTo(const TCallableType& typeToCompare, bool ign
     if (IsMergeDisabled0 != typeToCompare.IsMergeDisabled0)
         return false;
 
-    if (ArgumentsCount != typeToCompare.ArgumentsCount)
+    if (ArgumentsCount + typeToCompare.OptionalArgs != typeToCompare.ArgumentsCount + OptionalArgs)
```

Comment:
OptionalArgs are included into ArgumentsCount

Reply:
Sure. This condition is equivalent to `ArgumentsCount - OptionalArg != typeToCompare.ArgumentsCount - typeToCompare.OptionalArgs`

Reply:
it can be relaxed to
ArgumentsCount - OptionalArgs  >= typeToCompare.ArgumentsCount - typeToCompare.OptionalArgs && 
ArgumentsCount <= typeToCompare.ArgumentsCount

Reply:
for example Callable(T1,[T2,T3]) is compatible to Callable([T1,T2,T3,T4,T5])

Reply:
> for example Callable(T1,[T2,T3]) is compatible to Callable([T1,T2,T3,T4,T5])

For this example we will fail on `arg->IsConvertableTo` for the first arg.